### ISSUE TITLE
[BUG] MongoDB error-handling

### ISSUE THREAD

**Issue by vitasenka** [tomassirio/ListBot#154]
The bot replies with success message even if MongoDB server is not available.

How to produce the bug:
1. Start MongoDB server.
2. Start the bot.
3. Shutdown the MongoDB server.
4. Use bot commands.

![catchBug](https://user-images.githubusercontent.com/64737770/97445043-0c4f6d80-1942-11eb-9940-1897e807261d.jpg)

**Comment by RazCrimson** [comment]
Yes, I can reproduce this. The reply from bot takes some time(30seconds) which is probably when the request to DB times out.

**Comment by RazCrimson** [comment]
@rgroves We can stop the bot when the DB gets disconnected the connection times out?

**Comment by rgroves** [comment]
LIstBot needs better error handling. In some places, especially in most commands when interacting with the database, it's non-existent.

I think it would be good to have two things done in starting to address this issue, and this probably needs some further examination of the codebase to see exactly what needs doing, but in general:

1. Allow user to make connection timeout configurable—meaning they can lower the timeout value by overriding in the .env file. (this will make testing easier as well, so you won't need to wait the full 30 second default timeout).

2. Add try...catch blocks where needed in the commands so that we can capture, log, and report back a clear message to bot users when there is an issue.

This is something that definitely should be addressed.
